includes/internal/emails/class-email-list-table.php

```diff
@@ -230,4 +240,108 @@ private function get_row_actions( $post ): array {
 		 */
 		return apply_filters( 'sensei_email_list_row_actions', $actions, $post, $this );
 	}
+
+	/**
+	 * Render the bulk actions.
+	 *
+	 * @param string $which The location of the bulk actions: 'top' or 'bottom'.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The bulk actions HTML.
+	 */
+	public function bulk_actions( $which = '' ) {
+		if ( 'top' !== $which ) {
+			return '';
+		}
+
+		/**
+		 * Filter the bulk actions displayed on the email list.
+		 *
+		 * @since $$next-version$$
+		 * @hook sensei_email_list_bulk_actions
+		 *
+		 * @param {array} $bulk_actions The bulk actions.
+		 *
+		 * @return {array}
+		 */
+		$bulk_actions = apply_filters(
+			'sensei_email_list_bulk_actions',
+			[
+				'bulk-disable-email' => __( 'Disable', 'sensei-lms' ),
+				'bulk-enable-email'  => __( 'Enable', 'sensei-lms' ),
+			]
+		);
+
+		$output = '<label for="bulk-action-selector-top" class="screen-reader-text">' . __( 'Bulk Email Actions', 'sensei-lms' ) . '</label>'
+		. '<select name="action" id="bulk-action-selector-top">'
+		. '<option value="-1">' . __( 'Bulk Email Actions', 'sensei-lms' ) . '</option>';
+
+		foreach ( $bulk_actions as $key => $value ) {
+			$output .= '<option value="' . esc_attr( $key ) . '">' . $value . '</option>';
+		}
+
+		$output .= '</select>';
+
+		$output .= get_submit_button( __( 'Select Action', 'sensei-lms' ), 'action', '', false, array( 'id' => 'doaction' ) );
+
+		echo wp_kses(
+			$output,
+			[
+				'select' => [
+					'id'    => [],
+					'name'  => [],
+					'class' => [],
+				],
+				'option' => [
+					'value' => [],
+					'class' => [],
+				],
+				'input'  => [
+					'type'  => [],
+					'id'    => [],
+					'name'  => [],
+					'value' => [],
+					'class' => [],
+				],
+				'label'  => [
+					'for'   => [],
+					'class' => [],
+				],
+			]
+		);
+
+		return $output;
+	}
+
+	/**
+	 * Display table content wrapped inside a form
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return void
+	 */
+	public function display() {
+		$allowed_html = [
+			'form'  => [
+				'id'     => [],
+				'action' => [],
+				'method' => [],
+			],
+			'input' => [
+				'type'  => [],
+				'name'  => [],
+				'value' => [],
+			],
+		];
+
+		echo wp_kses( '<form id="posts-filter" action="' . esc_url( admin_url( 'edit.php' ) ) . '" method="get">', $allowed_html );
+		parent::display();
+		echo wp_kses(
+			'<input type="hidden" name="post_type" value="' . Email_Post_Type::POST_TYPE . '">' .
+			wp_nonce_field( 'sensei_email_bulk_action', '_wpnonce', true, false ) .
+			'</form>',
+			$allowed_html
+		);
```

Comment:
I don't think you need `wp_kses` here. You can just escape the attributes and that's enough.

```suggestion
		echo '<form id="posts-filter" action="' . esc_url( admin_url( 'edit.php' ) ) . '" method="get">';
		parent::display();
		echo '<input type="hidden" name="post_type" value="' . esc_attr( Email_Post_Type::POST_TYPE ) . '">';
		wp_nonce_field( 'sensei_email_bulk_action' );
		echo '</form>';
		);
```

Reply:
Thanks, updated here https://github.com/Automattic/sensei/pull/6552/commits/f6ab972411229b3fddd16d5177be81eeacc8b56d